src/InputField.tsx

```diff
@@ -4,17 +4,18 @@ import * as React from "react";
 import { InputProps } from "antd/lib/input";
 import { FormikFieldProps } from "./FieldProps";
 
-export const InputField = (props: FormikFieldProps & InputProps) => (
-  <Field name={props.name} validate={props.validate}>
-    {({ field }: FieldProps) => {
-      return (
-        <Input
-          value={field.value}
-          onChange={field.onChange}
-          onBlur={field.onBlur}
-          {...props}
-        />
-      );
-    }}
+export const InputField = (
+  { name, validate, ...restProps }: FormikFieldProps & InputProps
+) => (
+  <Field name={name} validate={validate}>
+    {({ field: { name: fieldName, value, onChange, onBlur } }: FieldProps) => (
+      <Input
+        name={fieldName}
```

Comment:
Same as checkbox, its not mentioned in the offical docs. I guess its the underlying html name attribute that get exposed (I can see it also with typescript). So I guess it makes sense to set it.
What about just using the _name_ prop from line 8, so the renaming in line 11 can be removed (I think it is much more complicated to read this way).

Reply:
Initially I agree with you that using the name from line 8 would eliminate the need for the name destructuring on line 11. However, I opted to specifically use the name provided by the Field component itself just in case Formik's Field modifies the name in any way; that way we know for sure that the name Formik is using is exactly the one it provides. As an added benefit this allows us to separate the User provided props from the Formik provided props cleanly.

If you feel this is ultimately overkill, I don't mind patching every instance of fieldName. Just wanted to give you the reason as to why I did it this way.

Reply:
After thinking a little about it I would now even remove setting the name property (for all components) because of the following reasons

- the added functionality is not clear
- the readability decreased
- the intention of the code is a bit obfuscated (its difficult for the reader to check what the name thing is about, is the outer thing important or the inner one)
- simplicity
- doesn't relate to the functionality proposed by this pull request

The last two are very important. I value simplicity a lot (imho its one of the most important "metrics" of quality software).
Also I don't know about what it actually means setting this value. If it turns out there is some problem with it, then I would have liked to have captured this change in its own commit/pull request.

So pls, remove setting the *name* property on the antd components altogether. If you need to to set the value, you can open another pull request.

Reply:
I can respect that.

I'd love to completely remove the name from the Antd components but some of them are required in order for the onChange to work properly.

If I did what you ask, I'd be committing broken components.

Reply:
Ok weird. Imho it should be possible to use antd components fully controlled (passing value and onChange) without setting _name_

I will accept then and double check later.